cloud/openssh/sshshim/mutagen.go

```diff
@@ -0,0 +1,89 @@
+package sshshim
+
+import (
+	"bytes"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.jetpack.io/devbox/cloud/mutagenbox"
+	"go.jetpack.io/devbox/debug"
+)
+
+// returns true if a liveVM is found, OR sshArgs were connecting to a server that is not a devbox-VM.
+// returns false iff the sshArgs were connecting to a devbox VM AND a deadVM is found.
+func EnsureLiveVMOrTerminateMutagenSessions(sshArgs []string) (bool, error) {
+	vmAddr := vmAddressIfAny(sshArgs)
+
+	debug.Log("Found vmAddr: %s", vmAddr)
+	if vmAddr == "" {
+		// We support the no Vm scenario, in case mutagen ssh-es into another server
+		// TODO savil. Revisit the no VM scenario if we can control the mutagen daemon for devbox-only
+		// syncing via MUTAGEN_DATA_DIRECTORY.
+		return true, nil
+	}
+
+	if isActive, err := checkActiveVM(vmAddr); err != nil {
+		return false, errors.WithStack(err)
+	} else if !isActive {
+		debug.Log("terminating mutagen session for vm: %s", vmAddr)
+		// If no vm is active, then we should terminate the running mutagen sessions
+		return false, terminateMutagenSessions(vmAddr)
+	}
+	return true, nil
+}
+
+func terminateMutagenSessions(vmAddr string) error {
+	username, hostname, found := strings.Cut(vmAddr, "@")
+	if !found {
+		hostname = username
+	}
+	machineID, _, found := strings.Cut(hostname, ".")
+	if !found {
+		return errors.Errorf(
+			"expected to find a period (.) in hostname (%s), but did not. "+
+				"For completeness, VmAddr is %s", hostname, vmAddr)
+	}
+
+	return mutagenbox.TerminateSessionsForMachine(machineID, nil /*env*/)
+}
+
+func checkActiveVM(vmAddr string) (bool, error) {
+
+	cmd := exec.Command("ssh", vmAddr, "echo 'alive'")
+
+	var bufErr, bufOut bytes.Buffer
+	cmd.Stderr = &bufErr
+	cmd.Stdout = &bufOut
+
+	err := cmd.Run()
+	if err != nil {
+		if err.Error() == "exit status 255" {
```

Comment:
If you unwrap this to an `exec.ExitError` you can check the exit code directly.

Reply:
ah yes, I had meant to do this, but forgot to update the code. Updated!